lib/pipelineFactory.js

```diff
@@ -0,0 +1,43 @@
+const clownface = require('clownface')
+const rdf = require('rdf-ext')
+const ns = require('./namespaces')
+const Pipeline = require('./pipeline')
+const jsLoader = require('./loader/ecmaScript')
+const templateStringLoader = require('./loader/ecmaScriptLiteral')
+const pipelineLoader = require('./loader/pipeline')
+const LoaderRegistry = require('./loader/registry')
+
+function pipelineNode (definition, iri) {
+  let node
+
+  if (iri) {
+    node = clownface(definition, rdf.namedNode(iri.value || iri.toString()))
+  } else {
+    node = clownface(definition).has(ns.rdf('type'), ns.p('Pipeline'))
+  }
+
+  if (!node.term) {
+    throw new Error('expected an existing IRI or a single Pipeline class in definition')
+  }
+
+  return node
+}
+
+function create (definition, { iri, basePath, context, objectMode, variables, additionalLoaders = [] } = {}) {
+  const defaultLoaders = [
+    jsLoader,
+    pipelineLoader,
+    templateStringLoader
+  ]
+
+  const loaders = [ ...defaultLoaders, ...additionalLoaders ]
```

Comment:
Loaders are a combinations of the defaults and whatever get's passed to the factory method.

Importantly, the constructor itself expects a registry object. That is to have the registry passed around between sub pipelines without unnecessary shenanigans.